File: DayOne/swap_glass.py
white_cup = "coffee"
transparent_cup = "tea"
transparent_glass = ""

print("Initial state")
print("White cup:", white_cup)
print("Transparent cup:", transparent_cup)
print("Transparent glass:", transparent_glass)

print()

print("Swapping!")
transparent_glass = transparent_cup
transparent_cup = white_cup
white_cup = transparent_glass

print()

print("End state")
print("White cup:", white_cup)
print("Transparent cup:", transparent_cup)
print("Transparent glass:", transparent_glass)

File: DayOne/swap_glass.swift
var whiteCup: String = "coffee"
var transparentCup: String = "tea"
var transparentGlass: String = ""

// Ignore following code
print("Initial State \nWhite Cup: \(whiteCup) \nTransparent Cup: \(transparentCup) \nTransparent Glass: \(transparentGlass)")

transparentGlass = transparentCup
transparentCup = whiteCup
whiteCup = transparentGlass

// Ignore following code
print("\nEnd State \nWhite Cup: \(whiteCup) \nTransparent Cup: \(transparentCup) \nTransparent Glass: \(transparentGlass)")